process/Style-guide.md

```diff
@@ -0,0 +1,250 @@
+# Style Guide
+
+## Rationale
+
+To keep the codebase maintainable and readable, all code is developed using a similar coding style. It ensures:
+
+* the code is easy to maintain and understand. 
+  * It's really important that the original developer able to come back to code 5 months later and still quickly understand what it's supposed to be doing. 
+  * For other people that want to contribute it is necessary for them to quickly understand the code.
+* that every developer knows to (largely) expect the same coding style.
+
+Having a unified style makes it much easier to maintain the our projects. That means that every developer should be able to make changes in any file in the log2timeline organization without worrying about different code styles. 
+
+# Project-specific deviations
+Some log2timeline projects systematically vary from this guide. Specifically:
+* [l2tscaffolder](https://github.com/log2timeline/l2tscaffolder) is not compatible with Python 2.
+* [dfTimewolf](https://github.com/log2timeline/l2tscaffolder) does not follow the naming rules, directly following the [Google Python Style Guide](https://google.github.io/styleguide/pyguide.html) instead.
+
+# Basis
+
+Log2Timeline's style is based on the the [Google Python Style Guide](https://google.github.io/styleguide/pyguide.html), with specific 
+variations and expansion spelled out in the sections below.
+
+## Compatibility
+All new log2timeline code should be compatible with both Python 3.4+ and Python 2.7+. The [Python 3 Guide](Python-3-Guide) has some more detail about compatibility issues, and the pylint configuration will also flag some issues.
+
+
+## Indentation
+
+* Indent your code blocks with 2 spaces (not 4 as in the style guide).
+* In the case of a hanging indent, use four spaces (according to the style guide).
+
+## Naming
+
+* Use full English words everywhere. For example, use Event not Evt and Description not Desc.Python-3-Guide.md
+* Acronyms and initialisms should be preserved, such as HTMLParser and not HtmlParser.
+* Use "cls" as the name of the class variable in preference to "klass"
+* Method and function names follow the following logic (overriding the Google Python Style Guide):
+
+
+Type | Public | Internal
+---- | ---- | ----
+Functions | **CapWords()** | **_CapWords()** (protected) and **__CapWords()** (private)
+
+
+## Linting
+
+log2timeline uses pylint 1.7.x to enforce some additional best practices to keep the source code more readable. These are:
+
+* Limit the maximum number of arguments for function or method to 10
+
+Other things to keep in mind when using Pylint:
+* Use textual pylint overrides e.g. "# pylint: disable=no-self-argument" instead of "# pylint: disable=E0213". For a list of overrides see: http://docs.pylint.org/features.html
+* The canonical pylint configuration for log2timeline project is generated by l2tdevtools. 
+
+
+### Function and method arguments
+
+* Prefix unused function or method arguments with `unused_`. 
```

Comment:
I think the usage of `_unused` is no longer accurate.

Reply:
Still relevant as far I can see...

Reply:
What I've noticed with the enforcement of matching docstrings Args is that it only applies to args and kwargs

Reply:
OK, this might be more confusing than useful - removed.